lib/most_frequent_integer.rb

```diff
@@ -2,6 +2,37 @@
 # frequent integer. These integers might be positive or negative. If no most-frequent integer
 # exists, return nil.
 
-def find_most_frequent_integer(arr)
-  # put your code here
+#def find_most_frequent_integer(arr)
+#    num ={
+#  num = arr.max_by{|k,v| v}
+#  return num.first
+#end
+
+def create_collection_of_numbers(nums)
+  h={}
+  nums.each do |n|
+    if h.has_key?(n) == false
+      h[n]=1
+    elsif h.has_key?(n) == true
+      h[n]= h[n] +1
+    end
+  end
+  return h
+end
+
+def find_most_frequent_integer(nums)
+  h = create_collection_of_numbers(nums)
+  #arr = []
+  max = h.values.max
+  Hash[h.select { |k, v| v == max}]
+  if h.length == 1
```

Comment:
I can see why you might think `h.length == 1` is correct, but the problem is that `h` never actually changed. `h` is still the collection of the numbers with their frequencies. And it has _all_ the numbers in it–not just the ones with the max value.

When you ran `Hash[h.select { |k, v| v == max}]`, you got a result back that was the filtered Hash with only those key/value pairs which have a value of the max.

But you never stored that new hash anywhere. It's _that_ hash that you want to check to see if its length == 1.